src/components/TypingTest/TyingBox.tsx

```diff
@@ -0,0 +1,73 @@
+import { Dispatch, SetStateAction, useMemo } from "react";
+import Character from "./Character";
+import { KBLayoutType } from "../../enums/KBLayout";
+
+type Props = {
+  sentence: string;
+  userInput: string;
+  setUserInput: Dispatch<SetStateAction<string>>;
+  sourceLayout: KBLayoutType;
+  targetLayout: KBLayoutType;
+};
+
+function TypingBox(props: Props) {
+  const { sentence, userInput, setUserInput, sourceLayout, targetLayout } =
+    props;
+
+  function handleKeyDown(event: { key: string }) {
+    if (event.key == "Backspace") {
```

Comment:
This is too edge-case filled of a method to use for this. We should be using a normal <input> (hidden) to take care of key presses and pass the value through to wherever
That way we only have to handle focusing the input when it needs to be

Reply:
Do we want to allow for punctuation?

Reply:
Well yeah, the sentences might have punctuation. The text entry doesn't need to be limited in any way afaik
Just need to pass through the characters. We don't need to concern ourselves with key events.